ls1intum/Artemis
# Improve registration: add more customization options

Fixes #2165

Fixes #2148

# Checklist

### Motivation and Context
More customization possibilities for other universities

### Description
* Course instructors can now add a message that students need to confirm
* System admins can now configure more options when it comes to the login screen and to the course registration 
* This PR also includes updates to server and client dependencies

### Steps for Testing
1. Log in to Artemis using different options in the `application-artemis.yml` file and check that everything works properly / is displayed correctly
2. Create / edit courses with student registration enabled and a confirmation message
3. Try to register for these courses with different accounts (following different patterns)
4. Customize more of the new settings in `application-artemis.yml` and verify that everything works properly
